### PR TITLE
Fix: remove incorrect assert in utils_align_ptr_up_size_down()

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Remove incorrect assert in `utils_align_ptr_up_size_down()`.
A pointer is aligned, but a size is only adjusted to the new pointer.
The size does not have to be aligned.

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Review by lukaszstolarczuk — COMMENTED** [review]
do you have a test for that? 😉

File: test/utils/utils_linux.cpp [hunk 1]
```diff
@@ -169,3 +169,19 @@ TEST_F(test, utils_open) {
     EXPECT_EQ(utils_file_open(NULL), -1);
     EXPECT_EQ(utils_file_open_or_create(NULL), -1);
 }
+
+TEST_F(test, utils_align_ptr_up_size_down) {
+    uintptr_t ptr = 0x4000;
```
Comment (KFilipek) [inline test/utils/utils_linux.cpp:174]: Missing header for uintptr_t (`cstdint` or `stdint.h`).

Comment (ldorau) [reply test/utils/utils_linux.cpp:175]: Done

**Review by bratpiorka — APPROVED** [review]

**Comment by ldorau** [comment]
@lukaszstolarczuk @KFilipek please re-review

**Review by lukaszstolarczuk — COMMENTED** [review]
LGTM 👍

**Review by KFilipek — APPROVED** [review]

**Pull request merged**